src/main/java/rape/brutal/gitstarship/ammunition/Railgun.java

```diff
@@ -5,14 +5,26 @@
 
 package rape.brutal.gitstarship.ammunition;
 
+import rape.brutal.gitstarship.IMagnetable;
+import rape.brutal.gitstarship.IStorageable;
+
 /**
  * Created by haze on 22.03.2017.
  */
 public class Railgun extends Gun {
 
-    private int cost;
-    private String name;
-    private int size;
+
+    public Railgun(int damage, String name, int size, int cost) {
+        super(damage, name, size, cost);
+    }
+
+
+    public int fire(IStorageable iStorageable) {
+        if (iStorageable instanceof IMagnetable) {
+            return damage;
```

Comment:
Wrong. Damage should vary rely to IMagnetable object parameters. From size, for example.